.NET/Microsoft.Recognizers.Text.NumberWithUnit/Extractors/BaseMergedUnitExtractor.cs

```diff
@@ -155,6 +156,15 @@ private void MergePureNumber(string source, List<ExtractResult> ers)
                     continue;
                 }
 
+                if (source.Length > numErs[i].Start + numErs[i].Length)
```

Comment:
Can you add a comment here explaining the purpose of this check?
Is there any perf impact?

Reply:
I just tested it and there is no performance drop.